python/pyspark/sql/connect/dataframe.py

```diff
@@ -1280,10 +1280,20 @@ def __getitem__(self, item: Union[int, str, Column, List, Tuple]) -> Union[Colum
         if isinstance(item, str):
             # Check for alias
             alias = self._get_alias()
-            if alias is not None:
-                return col(alias)
+            _col = alias if alias is not None else item
+            if _col == "*":
+                return Column(UnresolvedStar(unparsed_target=None))
+            elif _col.endswith(".*"):
+                return Column(UnresolvedStar(unparsed_target=_col))
             else:
-                return col(item)
+                if self._plan is None:
+                    raise Exception("Cannot analyze on empty plan.")
+                return Column(
+                    ColumnReference(
```

Comment:
is this the only place returning ColumnReference ?

Reply:
`functions.col` also return ColumnReference, but without plan_id

https://github.com/apache/spark/blob/6f376676ee21a228ca662fcfc12ff70d11acabfe/python/pyspark/sql/connect/functions.py#L204-L210

Reply:
will be great if we can reduce code duplication somehow

Reply:
like having a private version of `def col` which takes an extra plan id.

Reply:
nice, will update